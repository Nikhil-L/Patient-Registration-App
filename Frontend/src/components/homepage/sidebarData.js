export const SidebarData = [
    {
        title:"Patient Details",
        link:"/Display"
    },
    {
        title:"Register",
        link:"/Register"
    },
    {
        title:"Logout",
        link:"/"
    }
]