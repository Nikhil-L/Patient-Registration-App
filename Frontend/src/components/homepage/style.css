.heading{
    width: 100%;
    height: 100%;
    padding-top: 1px;
    padding-bottom: 1px;
    color: white;
    background-color: #434f5a;
    text-align: center;
}

.Sidebar{
    width: 200px;
    height: 100%;
    background-color: #2f4050;
}

.welcome{
    font-size: 130%;
    color: white;
    width:100%;
    height:100%;
    text-align: center;
    padding-top: 10px;
    padding-bottom: 50px;
}


.SidebarList{
    width: 100%;
    height: 434px;
    margin: 0;
    padding: 0;
}

.row{
    font-size: 130%;
    margin:0;
    padding-top: 15px;
    padding-bottom: 25px;
    width: 100%;
    color: white;
    text-align: center;
    list-style-type: none;
}



#title{
    padding-top: 20px;
    width: 100%;
    height: 40px;
}

.active{
    font-size: 110%;
    margin:0;
    padding-top: 15px;
    padding-bottom: 25px;
    width: 100%;
    background-color: black;
    color: white;
    text-align: center;
    list-style-type: none;
}

#title :hover{
    background-color: black;
}

li :active{
    background-color: black;
}

.row :hover{
    background-color: black;
}
