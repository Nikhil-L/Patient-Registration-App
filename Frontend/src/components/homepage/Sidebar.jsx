import React from "react";
import { SidebarData } from "./sidebarData";
import { Register } from "../register/Register";
import { Details } from "../details/Details";


/**
 * This component is used to render the sidebar to select options of either getting
 * patient details, registering a user or loging out
 */
export class Sidebar extends React.Component{

    /**
    * Initializes the props
    * @param {*} props 
    */
    constructor(props) {
        super(props);
    
        this.state = {
          title:"Patient Details",
          clicked:false,
          data: SidebarData
        };

        this.handleClick = this.handleClick.bind(this);
    
    }

    /**
     * This is to handle the logic to perform on click of buttons
     * @param {*} title 
     */
    handleClick(title)
    {
        
        if(title === "Register")
        {    this.setState({clicked:true});
            this.setState({title:"Register"});
        }
        else{
            this.setState({clicked:false});
            this.setState({title:"Patient Details"});
        }
        if(title === "Logout")
        {
            window.location = "/";
            sessionStorage.clear();
        }
        
    }

     /**
     * This is used to render the component to the webpage
     * @returns a html code 
     */
    render(){
        return(
            <div>
                {this.state.clicked?<Register/>:<Details/>}
                <div className="Sidebar">
                    <div className="welcome">Welcome</div>
                    <ul className="SidebarList">
                        {this.state.data.map((val, key) => {
                            return(     
                                <li 
                                    className="row"
                                    name={val.title}
                                    key={key} 
                                    onClick={()=>this.handleClick(val.title)}>
                                    <div id="title"
                                        className={this.state.title === val.title?"active":""}>{val.title}</div>
                                </li>
                            );
                        })}
                    </ul>
                    
                </div>
            </div>
        );
    }
}

