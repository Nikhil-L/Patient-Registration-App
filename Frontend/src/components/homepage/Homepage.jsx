import React from "react";
import './style.css';
import { Sidebar } from "./Sidebar";
import  { Redirect } from 'react-router-dom';

/**
 * This component renders the homepage of the webpage
 */
export class Homepage extends React.Component{

  /**
   * Initializes the props
   * @param {*} props 
   */
  constructor(props) {
      super(props);
  
      this.state = {
        input: {},
        errors: {}
      };
  
    }

       /**
  * This is used to render the component to the webpage
  * @returns a html code 
  */
  render(){
      let isAuthenticated = sessionStorage.getItem("isAuthenticated")
      if(isAuthenticated)
      {
        return(
          <div>
              <div className="heading">
                  <h1>Patient Registration</h1>
              </div >
              <Sidebar />
          </div>
        );
      }
      else{
        return <Redirect to='/'/>
      }
      
  }
}