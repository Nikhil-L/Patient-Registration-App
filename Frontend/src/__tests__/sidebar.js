import React from "react";
import ReactDOM from 'react-dom'
import {Sidebar} from "../components/homepage/Sidebar"
import renderer from "react-test-renderer"
import {shallow} from "enzyme"

/**
 * This is to test the Sidebar component
 */
describe("Sidebar test", ()=>{

  /**
   * Snapshot testing
   */
  it("matches a snapshot", ()=>{
    const tree = renderer.create(<Sidebar/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  /**
   * event testing
   */
  it("renders without crashing", ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<Sidebar/>, div);
  })
  
  /**
   * event testing
   */
  it('event testing', ()=>{
    let wrapper = shallow(<Sidebar/>);
    wrapper.find({name:"Register"}).simulate('click')
    wrapper.find({name:"Patient Details"}).simulate('click')
    wrapper.find({name:"Logout"}).simulate('click')
  })
})


