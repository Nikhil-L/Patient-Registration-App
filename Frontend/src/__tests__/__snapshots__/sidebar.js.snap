// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar test matches a snapshot 1`] = `
<div>
  <div>
    <div
      className="search"
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <button
        className="btn"
        onClick={[Function]}
        type="submit"
        value="submit"
      >
        Search
      </button>
    </div>
    <div
      className="details"
    >
      <table
        className="tbody"
      >
        <thead>
          <tr>
            <th>
              FIRST NAME
            </th>
            <th>
              LAST NAME
            </th>
            <th>
              GENDER
            </th>
            <th>
              REGISTRATION DATE
            </th>
            <th>
              DATE OF BIRTH
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Arun
            </td>
            <td>
              Kumar
            </td>
            <td>
              Male
            </td>
            <td>
              2021-06-05
            </td>
            <td>
              1976-05-27
            </td>
          </tr>
          <tr>
            <td>
              Varun
            </td>
            <td>
              Kumar
            </td>
            <td>
              Male
            </td>
            <td>
              2021-06-05
            </td>
            <td>
              1976-04-23
            </td>
          </tr>
          <tr>
            <td>
              Arun
            </td>
            <td>
              Kumar
            </td>
            <td>
              Male
            </td>
            <td>
              2021-06-05
            </td>
            <td>
              1976-05-27
            </td>
          </tr>
          <tr>
            <td>
              Ranjan
            </td>
            <td>
              Sharma
            </td>
            <td>
              Male
            </td>
            <td>
              2021-06-05
            </td>
            <td>
              1976-05-27
            </td>
          </tr>
          <tr>
            <td>
              Charan
            </td>
            <td>
              R
            </td>
            <td>
              Male
            </td>
            <td>
              2021-05-07
            </td>
            <td>
              1994-07-17
            </td>
          </tr>
          <tr>
            <td>
              Narendra
            </td>
            <td>
              S
            </td>
            <td>
              Male
            </td>
            <td>
              2021-04-07
            </td>
            <td>
              1999-07-17
            </td>
          </tr>
          <tr>
            <td>
              Roopa
            </td>
            <td>
              S
            </td>
            <td>
              Female
            </td>
            <td>
              2021-03-06
            </td>
            <td>
              1989-05-07
            </td>
          </tr>
          <tr>
            <td>
              Nikhil
            </td>
            <td>
              L
            </td>
            <td>
              Male
            </td>
            <td>
              2021-06-04
            </td>
            <td>
              1999-03-09
            </td>
          </tr>
          <tr>
            <td>
              Raju
            </td>
            <td>
              R
            </td>
            <td>
              Male
            </td>
            <td>
              2021-03-06
            </td>
            <td>
              1989-05-07
            </td>
          </tr>
          <tr>
            <td>
              Charan
            </td>
            <td>
              R
            </td>
            <td>
              Male
            </td>
            <td>
              2021-05-07
            </td>
            <td>
              1994-07-17
            </td>
          </tr>
          <tr>
            <td>
              Apoorva
            </td>
            <td>
              Vasist
            </td>
            <td>
              Female
            </td>
            <td>
              2021-04-06
            </td>
            <td>
              1997-03-08
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="Sidebar"
  >
    <div
      className="welcome"
    >
      Welcome
    </div>
    <ul
      className="SidebarList"
    >
      <li
        className="row"
        name="Patient Details"
        onClick={[Function]}
      >
        <div
          className="active"
          id="title"
        >
          Patient Details
        </div>
      </li>
      <li
        className="row"
        name="Register"
        onClick={[Function]}
      >
        <div
          className=""
          id="title"
        >
          Register
        </div>
      </li>
      <li
        className="row"
        name="Logout"
        onClick={[Function]}
      >
        <div
          className=""
          id="title"
        >
          Logout
        </div>
      </li>
    </ul>
  </div>
</div>
`;
